### PR TITLE
Fix defects found by resumption_test

### PR DESCRIPTION
**Pull request by okozlovlux**
- fixed defect in resumption unit tests

Related to [APPLINK-27757](https://adc.luxoft.com/jira/browse/APPLINK-27757), [APPLINK-27153](https://adc.luxoft.com/jira/browse/APPLINK-27153)

### CODE REVIEW

**Comment by okozlovlux** [comment]
@LevchenkoS @anosach-luxoft @vlantonov @pvvasilev @AByzhynar @wolfylambova22 @Kozoriz @VVeremjova please review

File: src/components/resumption/test/last_state_test.cc [hunk 1]
```diff
@@ -31,31 +31,43 @@
...
   static void SetUpTestCase() {
     file_system::DeleteFile(kAppInfoStorageFile);
     file_system::RemoveDirectory(kAppStorageFolder, false);
   }
   void SetUp() OVERRIDE {
     ASSERT_TRUE(file_system::CreateFile(app_info_dat_file_));
+
+    const std::vector<uint8_t> char_vector_pdata(kJsonStr.begin(),
+                                                 kJsonStr.end());
+    DCHECK(file_system::Write(kAppInfoStorageFile, char_vector_pdata));
```
Comment (LevchenkoS) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux, maybe it would be better to use here `ASSERT_TRUE`?

**Comment by LevchenkoS** [comment]
@okozlovlux, reviewed.

**Comment by VVeremjova** [comment]
@okozlovlux Reviewed

**Comment by vlantonov** [comment]
@okozlovlux Reviewed

**Comment by pvvasilev** [comment]
@okozlovlux Reviewed.

File: src/components/resumption/test/last_state_test.cc (hunk 1 above)
Comment (okozlovlux) [inline src/components/resumption/test/last_state_test.cc]: @LevchenkoS , fixed ecfd05d

File: src/components/resumption/test/last_state_test.cc [hunk 2]
```diff
@@ -31,31 +31,43 @@
...
 
-using namespace ::resumption;
+using ::utils::SharedPtr;
+using ::utils::MakeShared;
 
 const std::string kAppStorageFolder = "app_storage_folder";
-const std::string kAppInfoStorageFile = "app_info_storage";
+const std::string kAppInfoStorageFile = "app_info_storage.dat";
+const std::string kJsonStr = "{\"firstName\":\"name\"}";
+
```
Comment (AByzhynar) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux I would wrap these global consts to nameless namespace

Comment (okozlovlux) [inline src/components/resumption/test/last_state_test.cc]: @AByzhynar , fixed d8bb739

File: src/components/resumption/test/last_state_test.cc [hunk 3]
```diff
@@ -64,55 +76,78 @@ class LastStateTest : public ::testing::Test {
...
+  ::utils::SharedPtr<resumption::LastState> last_state_;
+
+  void CompactJson(std::string& str) {
+    Json::Value root;
+    Json::Reader reader;
+    reader.parse(str, root);
+    Json::FastWriter writer;
+    str = writer.write(root);
+    if (str[str.size() - 1] == '\n') {
+      str.erase(str.size() - 1, 1);
```
Comment (AByzhynar) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux What do you think bout the following:
```
string::iterator last_symbol = str.rbegin();
if ('\n' == *last_symbol) {
`str.erase (last_symbol)` 
}
```
Or even can be without additional variable `last_symbol`

File: src/components/resumption/test/last_state_test.cc [hunk 4]
```diff
@@ -64,55 +76,78 @@ class LastStateTest : public ::testing::Test {
...
   utils::json::JsonValue tcp_adapter_info =
       dictionary["TransportManager"]["TcpAdapter"];
-  EXPECT_EQ("{\n   \"devices\" : \"bluetooth_device\"\n}\n",
-            bluetooth_info.ToJson(true));
-  EXPECT_EQ(
-      "{\n   \"devices\" : {\n      \"name\" : \"test_device\"\n   }\n}\n",
-      tcp_adapter_info.ToJson(true));
+  std::string str = bluetooth_info.ToJson(true);
+  CompactJson(str);
+  EXPECT_EQ("{\"devices\":\"bluetooth_device\"}", str);
```
Comment (AByzhynar) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux May be use `const` variables instead ?

File: src/components/resumption/test/last_state_test.cc (hunk 3 above)
Comment (AByzhynar) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux Sorry. You can't do it this way as erase requires forward iterator. My mistake. But if you like such approach you can use base() to convert reverse_iterator to forward_iterator

Comment (okozlovlux) [inline src/components/resumption/test/last_state_test.cc]: @AByzhynar , that is ok.  already checked that I cann't do in such way as you wrote

File: src/components/resumption/test/last_state_test.cc (hunk 4 above)
Comment (okozlovlux) [inline src/components/resumption/test/last_state_test.cc]: @AByzhynar , fixed 56f57b6

**Comment by Kozoriz** [comment]
@okozlovlux , you changed part of logic in tests. Is this logic correct for all 3 platforms?

**Comment by okozlovlux** [comment]
@Kozoriz , I have checked it for all 3 platforms.

**Comment by Kozoriz** [comment]
@okozlovlux Reviewed

File: src/components/resumption/test/last_state_test.cc (hunk 3 above)
Comment (AByzhynar) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux Anyway I think that STL usage is more close to C++ development. SO I propose to use this version: 
`if ('\n' == *(s.end() -1)) {
    s.erase(s.end() -1);
}`

Comment (okozlovlux) [inline src/components/resumption/test/last_state_test.cc]: @AByzhynar , done in 97426d2

**Comment by wolfylambova22** [comment]
@okozlovlux Reviewed,

File: src/components/resumption/test/last_state_test.cc [hunk 5]
```diff
@@ -31,31 +31,45 @@
...
 
 #include <string>
+
 #include "gtest/gtest.h"
 #include "resumption/last_state.h"
 #include "utils/file_system.h"
+#include "utils/shared_ptr.h"
+#include "utils/make_shared.h"
+#include <json/writer.h>
+#include <json/reader.h>
```
Comment (anosach-luxoft) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux you cannot use `jsoncpp` directly. What about Qt build?

File: src/components/resumption/test/last_state_test.cc [hunk 6]
```diff
@@ -64,55 +78,81 @@ class LastStateTest : public ::testing::Test {
...
   utils::json::JsonValue bluetooth_info =
       dictionary["TransportManager"]["BluetoothAdapter"];
   utils::json::JsonValue tcp_adapter_info =
       dictionary["TransportManager"]["TcpAdapter"];
-  EXPECT_EQ("{\n   \"devices\" : \"bluetooth_device\"\n}\n",
-            bluetooth_info.ToJson(true));
-  EXPECT_EQ(
-      "{\n   \"devices\" : {\n      \"name\" : \"test_device\"\n   }\n}\n",
-      tcp_adapter_info.ToJson(true));
+  std::string str = bluetooth_info.ToJson(true);
```
Comment (anosach-luxoft) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux `str` not very good name

File: src/components/resumption/test/last_state_test.cc (hunk 5 above)
Comment (okozlovlux) [inline src/components/resumption/test/last_state_test.cc]: @anosach-luxoft , fixed 	eae6046
Checked all 3 platforms

File: src/components/resumption/test/last_state_test.cc (hunk 6 above)
Comment (okozlovlux) [inline src/components/resumption/test/last_state_test.cc]: @anosach-luxoft , fixed 	eae6046

File: src/components/resumption/test/last_state_test.cc [hunk 7]
```diff
@@ -64,55 +78,77 @@ class LastStateTest : public ::testing::Test {
   const std::string empty_dictionary_;
   const std::string app_info_dat_file_;
 
-  resumption::LastState last_state_;
+  ::utils::SharedPtr<resumption::LastState> last_state_;
+
+  void CompactJson(std::string& str) {
+    str.erase(remove_if(str.begin(), str.end(), isspace), str.end());
```
Comment (AByzhynar) [inline src/components/resumption/test/last_state_test.cc]: @okozlovlux Are you sure that all spaces should be deleted?

Comment (okozlovlux) [inline src/components/resumption/test/last_state_test.cc]: @AByzhynar . Yes. In previuos realization also all spaces, `\n`, `\t' were deleteing.

**Comment by anosach-luxoft** [comment]
Reviewed

**Comment by AByzhynar** [comment]
Reviewed

**Pull request merged**